config_src/nuopc_driver/mom_cap_methods.F90

```diff
@@ -828,4 +765,193 @@ subroutine State_SetExport(state, fldname, isc, iec, jsc, jec, input, ocean_grid
 
 end subroutine State_SetExport
 
+subroutine state_diagnose(State, string, rc)
+
+  ! ----------------------------------------------
+  ! Diagnose status of State
+  ! ----------------------------------------------
+
+  type(ESMF_State), intent(in)  :: state
+  character(len=*), intent(in)  :: string
+  integer         , intent(out) :: rc
+
+  ! local variables
+  integer                         :: i,j,n
+  type(ESMf_Field)                :: lfield
+  integer                         :: fieldCount, lrank
+  character(ESMF_MAXSTR) ,pointer :: lfieldnamelist(:)
+  real(ESMF_KIND_R8), pointer     :: dataPtr1d(:)
+  real(ESMF_KIND_R8), pointer     :: dataPtr2d(:,:)
+  character(len=*),parameter      :: subname='(state_diagnose)'
+  character(len=ESMF_MAXSTR)      :: msgString
+  ! ----------------------------------------------
+
+  call ESMF_StateGet(state, itemCount=fieldCount, rc=rc)
+  if (chkerr(rc,__LINE__,u_FILE_u)) return
+  allocate(lfieldnamelist(fieldCount))
+
+  call ESMF_StateGet(state, itemNameList=lfieldnamelist, rc=rc)
+  if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+  do n = 1, fieldCount
+
+     call ESMF_StateGet(state, itemName=lfieldnamelist(n), field=lfield, rc=rc)
+     if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+     call field_getfldptr(lfield, fldptr1=dataPtr1d, fldptr2=dataPtr2d, rank=lrank, rc=rc)
+     if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+     if (lrank == 0) then
+        ! no local data
+     elseif (lrank == 1) then
+        if (size(dataPtr1d) > 0) then
+           write(msgString,'(A,3g14.7,i8)') trim(string)//': '//trim(lfieldnamelist(n)), &
+                minval(dataPtr1d), maxval(dataPtr1d), sum(dataPtr1d), size(dataPtr1d)
+        else
+           write(msgString,'(A,a)') trim(string)//': '//trim(lfieldnamelist(n))," no data"
+        endif
+     elseif (lrank == 2) then
+        if (size(dataPtr2d) > 0) then
+           write(msgString,'(A,3g14.7,i8)') trim(string)//': '//trim(lfieldnamelist(n)), &
+                minval(dataPtr2d), maxval(dataPtr2d), sum(dataPtr2d), size(dataPtr2d)
+        else
+           write(msgString,'(A,a)') trim(string)//': '//trim(lfieldnamelist(n))," no data"
+        endif
+     else
+        call ESMF_LogWrite(trim(subname)//": ERROR rank not supported ", ESMF_LOGMSG_ERROR)
+        rc = ESMF_FAILURE
+        return
+     endif
+     call ESMF_LogWrite(trim(msgString), ESMF_LOGMSG_INFO)
+  enddo
+
+  deallocate(lfieldnamelist)
+
+end subroutine state_diagnose
+
+!===============================================================================
+
+subroutine field_getfldptr(field, fldptr1, fldptr2, rank, abort, rc)
```

Comment:
Please document this subroutine, including its arguments, using Doxygen style. You can subroutine mom_import as an example.